jax/_src/pjit.py

```diff
@@ -2410,6 +2412,37 @@ def _pjit_state_discharge_rule(
   return new_invals, out_vals
 state_discharge.register_discharge_rule(pjit_p)(_pjit_state_discharge_rule)
 
+def _pjit_edtype_rule(ctx: jaxpr_passes.ResolveEdtypesContext,
+                          *args,
+                          name, jaxpr: core.ClosedJaxpr, in_shardings,
+                          out_shardings, in_layouts, out_layouts, resource_env,
+                          donated_invars, keep_unused, inline
+                           ):
+  physical_jaxpr = jaxpr_passes.resolve_edtypes_jaxpr(jaxpr)
+  in_shardings = [
+    sharding_impls.physical_sharding(a, s)
+    if a is not core.abstract_token and dtypes.issubdtype(a.dtype, dtypes.extended)
+    else s
+    for s, a in zip(in_shardings, ctx.avals_in)
+  ]
+  out_shardings = [
+    sharding_impls.physical_sharding(a, s)
+    if a is not core.abstract_token and dtypes.issubdtype(a.dtype, dtypes.extended)
+    else s
+    for s, a in zip(out_shardings, ctx.avals_out)
+  ]
```

Comment:
Also please raise NotImplementedError if `in_layout is not None` or `out_layouts is not Nont`

Reply:
Done.